src/modules/gcm/gcm-server.js

```diff
@@ -29,130 +32,116 @@ if (config.gcm.senderId) {
 	log.info('GCM module not enabled');
 }
 
-function sendStanza(changes, entity) {
-	// console.log('gcm server is here')
+async function sendStanza(changes, entity) {
+
 	if (entity.type === Constants.TYPE_THREAD) {
 		if (!entity.createTime || entity.createTime !== entity.updateTime) {
 			log.info('not new thread: ', entity);
 			return;
 		}
-		const counter = new Counter();
 		let room = changes.entities[entity.parents[0]];
 		if (!room || !room.name) {
-			counter.inc();
-			cache.getEntity(entity.parents[0], (e, roomObj) => {
-				room = roomObj;
-				counter.dec();
-			});
+			room = await getEntityAsync(entity.parents[0]);
 		}
-		// console.log("sdjkfhjd g: ", entity)
-		counter.then(() => {
-			const title = room.name + ': ' + entity.creator + ' started a discussion',
-				urlLink = config.server.protocol + '//' + config.server.host + convertRouteToURL({
-					name: 'chat',
-					props: {
-						room: entity && entity.parents[0],
-						thread: entity && entity.id,
-					},
-				});
-
-			log.info('sending pushnotification for thread', entity, urlLink);
-			const pushData: Note = {
-				group: entity.id,
-				count: 1,
-				score: 10,
-				data: {
-					body: entity.name,
-					creator: entity.creator,
+
+		const title = room.name + ': ' + entity.creator + ' started a discussion';
```

Comment:
@aravindet We can't add tags here (might conflict with the text). We'll need to do on client side. Will do the changes in a separate PR.